lib/ansible/modules/windows/win_nssm.ps1

```diff
@@ -482,6 +484,126 @@ Function Nssm-Update-Dependencies
     }
 }
 
+Function Convert-HashToString
+{
+    <#
+    Convert a hash into a variable length string of KEY=VALUE seperated by spaces
+
+    Input:
+    {
+        key1: value1,
+        key2: value2
+    }
+
+    Output:
+    "key1=value1 key2=value2"
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [hashtable]$hash
+    )
+
+    return ($hash.GetEnumerator() | % { "$($_.Key)=$($_.Value)" }) -join " "
+}
+
+Function Convert-ArrayToHash
+{
+    <#
+    Converts an array with key=value elements into a hashtable
+
+    Input:
+    ["key1=value1", "key2=value2"]
+
+    Output:
+    {
+        key1: value1,
+        key2: value2
+    }
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [array]$array
+    )
+
+    $hash = @{}
+
+    foreach ($element in $array)
+    {
+        $hash = $hash + (ConvertFrom-StringData $element)
+    }
+    return $hash
+}
+
+Function Nssm-Update-Extra-Environment
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$name,
+        [Parameter(Mandatory=$true)]
+        [hashtable]$env_vars
+    )
+
+    $env_vars_string = Convert-HashToString $env_vars
+
+    $cmd = "get ""$name"" AppEnvironmentExtra"
+    $results = Nssm-Invoke $cmd
+
+    if ($LastExitCode -ne 0)
+    {
+        $result.nssm_error_cmd = $cmd
+        $result.nssm_error_log = "$results"
+        Throw "Error updating environment variables for service ""$name"""
+    }
+
+    $current_env_vars = Convert-ArrayToHash $results
+
+    <# 
+    Update the appEnvironmentExtra values if:
+    - The services current environment variables does not match the desired environment variables.
+    - env_vars is not an empty hashtable
+    #>
+    If (((Convert-HashToString $env_vars) -ne (Convert-HashToString $current_env_vars)) -and ($env_vars.Count -gt 0)) 
```

Comment:
You may have a problem with this, I don't believe Ansible keeps the order of a dict being passed in so this string can be different. I think you will need a function to iterate through the hashtable and compare if the keys and their values match in either table.